x/staking/genesis.go

```diff
@@ -4,18 +4,28 @@ import (
 	"context"
 	"fmt"
 
-	cmttypes "github.com/cometbft/cometbft/types"
 	gogotypes "github.com/cosmos/gogoproto/types"
 
 	"cosmossdk.io/x/staking/keeper"
 	"cosmossdk.io/x/staking/types"
 
-	cryptocodec "github.com/cosmos/cosmos-sdk/crypto/codec"
+	cryptotypes "github.com/cosmos/cosmos-sdk/crypto/types"
 	sdk "github.com/cosmos/cosmos-sdk/types"
 )
 
+// TODO: move this to sdk types and use this instead of comet types GenesisValidator
+// then we can do pubkey conversion in ToGenesisDoc
+//
+// this is a temporary work around to avoid import comet directly in staking
+type GenesisValidator struct {
```

Comment:
maybe revert this and backport the changes from the follow up PR

Reply:
it will be easier for the next backport if the changes in this branch stays close to main